### PR TITLE
Windows AMI Pin vs 2022 to older msvc release

### PR DESCRIPTION
**Pull request by atalman**
See https://github.com/pytorch/vision/pull/8492 for reference

Link taken from here: https://learn.microsoft.com/en-us/visualstudio/releases/2022/release-history#evergreen-bootstrappers

### CODE REVIEW

**Review by malfet — APPROVED** [review]
Would be nice to add a reference to VC++ version CUDA supports
Also, why it affects only vision but not the core? Should we perhaps update something there?

**Comment by atalman** [comment]
Very detailed info on this comment: https://github.com/pytorch/vision/pull/8492#issuecomment-2166893457

File: aws/ami/windows/scripts/Installers/Install-VS.ps1 [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
+# Where to find the links:
 #   - https://docs.microsoft.com/en-us/visualstudio/releases/2019/history#release-dates-and-build-numbers
 #   - https://learn.microsoft.com/en-us/visualstudio/releases/2022/release-history#fixed-version-bootstrappers
 # BuildTools from S3
 $VS_VERSION_major = [int] ${env:VS_VERSION}.split(".")[0]
+
 $VS_DOWNLOAD_LINK = "https://aka.ms/vs/$VS_VERSION_major/release/vs_BuildTools.exe"
+
+# Pin to older vs_buildtools version for cuda 11.8 and 12.1 builds
+# see issue: nerfstudio-project/nerfstudio#3157
```
Comment (ZainRizvi) [inline aws/ami/windows/scripts/Installers/Install-VS.ps1:11]:
```suggestion
# see issue: https://github.com/nerfstudio-project/nerfstudio/issues/3157
```

File: aws/ami/windows/scripts/Installers/Install-VS.ps1 [hunk 2]
```diff
@@ -1,10 +1,18 @@
...
+if ( ${env:VS_YEAR} -eq "2022" ) {
+    $VS_DOWNLOAD_LINK = "https://aka.ms/vs/17/release.ltsc.17.8/vs_buildtools.exe"
```
Comment (ZainRizvi) [inline aws/ami/windows/scripts/Installers/Install-VS.ps1:13]: Does the casing of BuildTools.exe need to change between the two links? (line 8 and line 13)

**Review by ZainRizvi — APPROVED** [review]

File: aws/ami/windows/scripts/Installers/Install-VS.ps1 (hunk 2 above)
Comment (atalman) [reply aws/ami/windows/scripts/Installers/Install-VS.ps1:13]: Line 8 is still needed for MSVC 2019

**Pull request merged**